utils/digital_interrupts.go

```diff
@@ -37,12 +76,19 @@ const ServoRollingAverageWindow = 10
 // reconfiguration within the same type.
 type ReconfigurableDigitalInterrupt interface {
 	board.DigitalInterrupt
-	Reconfigure(cfg DigitalInterruptConfig) error
+	Reconfigure(cfg PinConfig) error
 }
 
-// CreateDigitalInterrupt is a factory method for creating a BasicDigitalInterrupt type.
-func CreateDigitalInterrupt(cfg DigitalInterruptConfig) (ReconfigurableDigitalInterrupt, error) {
+// CreateDigitalInterrupt is a factory method for creating a specific DigitalInterrupt based
+// on the given config. If no type is specified, an error is returned.
+func CreateDigitalInterrupt(cfg PinConfig) (ReconfigurableDigitalInterrupt, error) {
 	i := &BasicDigitalInterrupt{}
+	//nolint:exhaustive
+	switch cfg.Type {
+	case PinInterrupt:
+	default:
+		return nil, fmt.Errorf("expected pin %v to be configured as %v, got %v instead", cfg.Name, PinInterrupt, cfg.Type)
+	}
```

Comment:
This can be done currently without configuration, `DigitalInterruptByName` creates a pin on the fly without the pin being named in the config, by using its number. We should remove this.

Reply:
> personally i think we should not have the rpiutils package, because its just an extra layer of complexity

I totally agree, if you'd like to look at how we got rid of digital interrupt types for genericlinux, here's the PR: https://github.com/viamrobotics/rdk/commit/b7283352f839161d2a532da712370fd5433a1c3c#diff-564f7b48d2f6dfdf53f9ee88bac84adcf7e6d6e573ce861d0c93f915063388c2

Reply:
Just to make it explicit, we talked offline - nothing to do for now, just pointing out how we can get rid of this struct completely.